src/arg_max.py

```diff
@@ -0,0 +1,50 @@
+from nada_dsl import *
+import nada_numpy as na
+
+DIM = 10
+
+# A party to keep track of the index of the result
+index_party = Party(name="index_party")
+
+# A party holding the public value of 1, used for addition
+one_party = Party(name="one_party")
+
+# The result (the index of the argmax)
+result = SecretInteger(Input(name="indx", party=index_party))
+
+# The current index of the loop. It is set to be a public value
+current_indx = PublicInteger(Input(name="current_indx", party=index_party))
+
+# A public value of 1
+one = PublicInteger(Input(name="public_one", party=one_party))
+
+def argmax(array: na.NadaArray):
+    global result, current_indx, one
```

Comment:
I'd remove the global variables, they don't offer much and might also be confusing.

Reply:
Fixed! 
I had to change the `argmax` header to `def argmax(array: na.NadaArray, index_party: Party)`